prime-router/src/test/kotlin/fhirengine/azure/FHIRConverterIntegrationTests.kt

```diff
@@ -0,0 +1,456 @@
+            "observation-mapping" to LookupTable("observation-mapping", emptyList())
+        )
+        val workflowEngine = WorkflowEngine
+            .Builder()
+            .metadata(metadata)
+            .settingsProvider(settings)
+            .databaseAccess(ReportStreamTestDatabaseContainer.testDatabaseAccess)
+            .build()
+        return FHIRFunctions(workflowEngine, databaseAccess = ReportStreamTestDatabaseContainer.testDatabaseAccess)
+    }
+
+    private fun createFHIRConverter(): FHIRConverter {
+        val settings = FileSettings().loadOrganizations(universalPipelineOrganization)
+        val metadata = UnitTestUtils.simpleMetadata
+        metadata.lookupTableStore += mapOf(
+            "observation-mapping" to LookupTable("observation-mapping", emptyList())
+        )
+        return FHIRConverter(
+            metadata,
+            settings,
+            ReportStreamTestDatabaseContainer.testDatabaseAccess,
+        )
+    }
+
+    private fun generateQueueMessage(report: Report, blobContents: String, sender: Sender): String {
+        return """
+            {
+                "type": "convert",
+                "reportId": "${report.id}",
+                "blobURL": "${report.bodyURL}",
+                "digest": "${BlobAccess.digestToString(BlobAccess.sha256Digest(blobContents.toByteArray()))}",
+                "blobSubFolderName": "${sender.fullName}",
+                "topic": "${sender.topic.jsonVal}",
+                "schemaName": "${sender.schemaName}" 
+            }
+        """.trimIndent()
+    }
+
+    @BeforeEach
+    fun beforeEach() {
+        mockkObject(QueueAccess)
+        every { QueueAccess.sendMessage(any(), any()) } returns Unit
+        mockkObject(BlobAccess)
+        every { BlobAccess getProperty "defaultBlobMetadata" } returns getBlobContainerMetadata()
+        mockkObject(BlobAccess.BlobContainerMetadata)
+        every { BlobAccess.BlobContainerMetadata.build(any(), any()) } returns getBlobContainerMetadata()
+        mockkConstructor(DatabaseLookupTableAccess::class)
+    }
+
+    @AfterEach
+    fun afterEach() {
+        unmockkObject(BlobAccess)
+        unmockkObject(QueueAccess)
+    }
+
+    private fun getBlobContainerMetadata(): BlobAccess.BlobContainerMetadata {
+        val blobConnectionString =
+            """DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=keydevstoreaccount1;BlobEndpoint=http://${azuriteContainer.host}:${
+                azuriteContainer.getMappedPort(
+                    10000
+                )
+            }/devstoreaccount1;QueueEndpoint=http://${azuriteContainer.host}:${
+                azuriteContainer.getMappedPort(
+                    10001
+                )
+            }/devstoreaccount1;"""
+        return BlobAccess.BlobContainerMetadata(
+            "container1",
+            blobConnectionString
+        )
+    }
+
+    // TODO convert between sender format and report form
```

Comment:
I'll add a ticket for this as well, so both formats don't need to be passed

Reply:
reminder to create ticket and then update this comment with the number